File: leetsync/2503-longest-subarray-with-maximum-bitwise-and/longest-subarray-with-maximum-bitwise-and.go
func longestSubarray(nums []int) int {
	maxLength, streak, maxValue := 0, 0, 0
	for _, num := range nums {
		if maxValue < num {
			maxValue = num
			streak = 0
			maxLength = 0
		}
		if maxValue == num {
			streak++
		} else {
			streak = 0
		}
		maxLength = max(maxLength, streak)
	}
	return maxLength
}

File: leetsync/2503-longest-subarray-with-maximum-bitwise-and/longest-subarray-with-maximum-bitwise-and.java
class Solution {
    public int longestSubarray(int[] nums) {
        int n = nums.length;
        int maxLength = 0;
        int maxValue = 0, streak = 0;
        for (int num : nums) {
            if (maxValue < num) {
                maxValue = num;
                streak = 0;
                maxLength = 0;
            }
            if (maxValue == num) {
                streak++;
            } else {
                streak = 0;
            }
            maxLength = Math.max(maxLength, streak);
        }
        return maxLength;
    }
}